contrib/generate-ds20.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python3
+# pylint: disable=invalid-name,missing-docstring
+#
+# Copyright (C) 2022 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+#
+# pylint: disable=consider-using-f-string
+
+import sys
+import argparse
+import configparser
+import base64
+from typing import List
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-b",
+        "--bufsz",
+        type=int,
+        help="Buffer size in bytes",
+    )
+    parser.add_argument("-i", "--instance-id", type=str, help="Device instance ID")
+    parser.add_argument("filename", action="store", type=str, help="Quirk filename")
+
+    args = parser.parse_args()
+    if not args.bufsz:
+        parser.print_help()
+        sys.exit(1)
+
+    config = configparser.ConfigParser()
+    config.optionxform = str
+    try:
+        config.read(args.filename)
+    except configparser.MissingSectionHeaderError:
+        print("not a quirk file")
+        sys.exit(1)
+
+    # fall back to the default if there is only one device in the quirk file
+    if not args.instance_id:
+        sections = config.sections()
+        if len(sections) != 1:
+            print("multiple devices found, use --instance-id to choose")
+            sys.exit(1)
+        args.instance_id = sections[0]
+
+    # create the smallest kv store possible
+    lines: List[str] = []
+    try:
+        for key in config[args.instance_id]:
+            if key in ["Children", "Inhibit", "Issue", "Children"]:
```

Comment:
you have Children twice, but also why don't you want this key?  I would think that composite devices could still benefit just the same with this.

Reply:
Yup, I guess you're right. Both children removed.